core/src/main/scala/cats/collections/Range.scala

```diff
@@ -125,8 +125,8 @@ final case class Range[A](val start: A, val end: A) {
     * Folds over the elements of the range from right to left; accumulates a value of type B
     * by applying the function f to the current value and the next element.
     */
-  def foldRight[B](s: B, f: (A, B) => B)(implicit discrete: Discrete[A], order: Order[A]): B =
-    reverse.foldLeft(s, (b: B, a: A) => f(a, b))(discrete.inverse, Order.reverse(order))
+  def foldRight[AA >: A, B](s: B, f: (AA, B) => B)(implicit discrete: Discrete[AA], order: Order[AA]): B =
```

Comment:
Really weird choice that f is not a separate parameter group here. Seems not idiomatic.